### PR TITLE
Ganglion Native BLE

### PR DESCRIPTION
**Pull request by retiutut**
Fixes https://github.com/OpenBCI/OpenBCI_GUI/issues/1080

### CODE REVIEW

**Comment by Andrey1994** [comment]
In brainflow everything is done, and its in the process of porting to the gui...

**Comment by philippitts** [comment]
In later versions of MacOS (generally MacOS 11+) there is an issue where Bluetooth permissions are not requested by the application. This prevents the embedded native BLE library from accessing the Bluetooth adapter and therefore Ganglion boards (and other Bluetooth devices) cannot be found.

This issue appears to be related to [application hardening requirements](https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution) introduced in MacOS 10.14.5. Most importantly, the JDK 8 version that ships with Processing 3.5.4 and earlier [is not notarized](https://github.com/adoptium/temurin-build/issues/1130) and therefore cannot request permissions from the OS.

A workaround to the issue is updating the JDK in Processing to the latest version of JDK 8. This can be done with the following:

```
curl -O -L --insecure https://github.com/adoptium/temurin8-binaries/releases/download/jdk8u352-b08/OpenJDK8U-jdk_x64_mac_hotspot_8u352b08.tar.gz
tar -xf OpenJDK8U-jdk_x64_mac_hotspot_8u352b08.tar.gz
mv jdk8u352-b08 /Applications/Processing.app/Content/PlugIns/jdk1.8.0_202.jdk
```

Then follow the instructions for [running from the IDE](https://docs.openbci.com/Software/OpenBCISoftware/GUIDocs/#running-the-openbci-gui-from-the-processing-ide).

Note that applications exported from Processing 3.5.4 do not appear to ask for Bluetooth permissions even if the JDK is updated before exporting. Processing 3.5.4 is no longer maintained so downstream fixes may exist in Processing 4.

File: OpenBCI_GUI/ControlPanel.pde [hunk 1]
```diff
@@ -1161,6 +1201,21 @@ class InterfaceBoxGanglion {
...
     }
 
+    private void createGanglionNativeBLEButton(String name, String text, boolean isToggled, int _x, int _y, int _w, int _h) {
+        protocolGanglionNativeBLE = createIFBGButton(name, text, isToggled, _x, _y, _w, _h);
+        protocolGanglionNativeBLE.onRelease(new CallbackListener() {
+            public void controlEvent(CallbackEvent theEvent) {
+                controlPanel.wifiBox.wifiList.items.clear();
+                controlPanel.bleBox.bleList.items.clear();
+                selectedProtocol = BoardProtocol.NATIVE_BLE;
+                //controlPanel.bleBox.refreshGanglionBLEList();
```
Comment (philippitts) [inline OpenBCI_GUI/ControlPanel.pde:1211]: Is this needed? Recommend uncommenting it if it's useful or removing it if it's not.

File: OpenBCI_GUI/OpenBCI_GUI.pde [hunk 2]
```diff
@@ -331,7 +332,7 @@ void settings() {
         win_w = 980;
         win_h = 580;
     }
-    size(win_w, win_h, P2D);
+    size(win_w, win_h);
```
Comment (philippitts) [inline OpenBCI_GUI/OpenBCI_GUI.pde:335]: Why the renderer change here? If this is a leftover from exploring Processing 4 upgrade paths, we should save it for a Processing 4 specific PR.

**Review by philippitts — CHANGES_REQUESTED** [review]

File: OpenBCI_GUI/OpenBCI_GUI.pde (hunk 2 above)
Comment (retiutut) [reply OpenBCI_GUI/OpenBCI_GUI.pde:335]: Correct. Will do.

File: OpenBCI_GUI/ControlPanel.pde (hunk 1 above)
Comment (retiutut) [reply OpenBCI_GUI/ControlPanel.pde:1211]: I'll re-evaluate this line. Thanks. This PR needs a few more UX passes from myself, and I'll address these requested changes also.

**Review by philippitts — APPROVED** [review]

**Comment by retiutut** [comment]
Wooo! Let's gooooo!

**Pull request merged**